src/functional/iterator/embedded.py

```diff
@@ -586,27 +594,31 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
+    return axis is None
+
+
 def get_ordered_indices(
     axes: Iterable[Axis], pos: Mapping[str, FieldIndex]
 ) -> tuple[FieldIndex, ...]:
-    assert all(axis.value in [*pos.keys()] for axis in axes)
-    return tuple(pos[axis.value] for axis in axes)
+    assert all(axis.value in [*pos.keys()] for axis in axes if axis is not None)
+    return tuple(pos[axis.value] if not _is_tuple_axis(axis) else slice(None) for axis in axes)
```

Comment:
```suggestion
    return slice(None) if _is_tuple_axis(axis) else tuple(pos[axis.value] for axis in axes)
```

Reply:
went with your other proposal for `_is_field_axis`